src/frontend/src/pages/ProjectDetailPage/ProjectViewContainer/BOM/BOMTable.tsx

```diff
@@ -93,13 +118,31 @@ const BOMTable: React.FC<BOMTableProps> = ({ setHideColumn, columns, materials,
         columns={columns as GridColumns<GridValidRowModel>}
         rows={rows.concat(materialsWithAssemblies.filter(isAssemblyOpen))}
         getRowClassName={(params) =>
-          `super-app-theme--${String(params.row.id).includes('assembly') ? 'assembly' : 'material'}`
+          `super-app-theme--${String(params.row.id).includes('assembly') ? `assembly assembly-id:${params.row.assemblyId}` : `material material-id:${params.row.materialId}`}`
```

Comment:
I don't like injecting the class like this. 

I think you can get the material ID (line 135, 140) using something like `rows.concat(materialsWithAssemblies.filter(isAssemblyOpen))[parseInt(event.currentTarget.dataset.rowindex || '0')].materialId;` where you index the row IDs with the index of the row (but refactor this so it's not one line)

This way you don't need to modify class name